### PR TITLE
MRG+1, ENH: Allow creating DigMontage for standard montages

### PR DESCRIPTION
**Pull request by massich**
This works on #6461, It finds a way to create DigMontages from standard montage kind.

```py
def read_standard_montage(kind: str) -> DigMontage
```

This could either be a new `read_standard_montage` or reuse `read_montage`.

### CODE REVIEW

File: mne/channels/tests/test_standard_montage.py [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+
+
+def test_easycap_M1():
+    """Test easycap_M1."""
+    EXPECTED_HEAD_SIZE = 1.  # 0.085
+
+    montage = read_standard_montage('easycap_M1')
+    eeg_loc = np.array([ch['r'] for ch in _get_dig_eeg(montage.dig)])
+
+    ## XXX Thats what I thought it was right. But substracting the center fails
```
Comment (massich) [inline mne/channels/tests/test_standard_montage.py]: @agramfort 	I don't understand why when creating [x,y,z] from [1, phi, theta] why is not center in 0. and if its not centered in 0 why when centering the norm is no longer 1.

Comment (agramfort) [reply mne/channels/tests/test_standard_montage.py]: I don't know. I would suggest to do a stupid for loop with all standard montages. Make sure you can pass them to set_montage that it makes sense using fsaverage anatomy. Make sure about thate set_montage guarantees that the DigMontage can be mapped to head (or is already in head).

**Comment by massich** [comment]
I think that `make_standard_montage` or `load_standard_montage` is a better name than `read_standard_montage`.

I like `make_standard_montage` (just for coorence with `make_dig_montage`, `make_dig_ponints` whatever... despite the fact that its actually a `loader`)

**Comment by agramfort** [comment]
+1 for make_standard_montage as it's not indeed a read function as it does
not take a fname as input

>

**Comment by massich** [comment]
This is the [rendered example](https://15205-1301584-gh.circle-artifacts.com/0/dev/auto_examples/visualization/plot_montage.html#sphx-glr-auto-examples-visualization-plot-montage-py) in the current stage. Only montages without fiducials should be rendered properly. But this is not the case. **standard**s and **mgh**s do render properly which means that those points are already in `'head'`. `easycap` is wierdly off, and `HydroCell` and `biosemi` are behaving as expected. 

Fixing all this.

**Comment by agramfort** [comment]
cool

go go @massich !

>

**Comment by massich** [comment]
There is something really funky that I don't understand. 

Even for something that is `montage.plot()` works i.e `'mgh60'` the points had not been making sense to me. In the following (from 1e41d1c4b) I'm plotting several things. The points read from the montage (in unk coord), in blue the transform version of the montage, and in white some points in 0.085 sphere 'cos the sphere uses `scale` rather than radious and is not calibrated in the plot. (but is useful to see where 0 origin is)

image

File: mne/channels/tests/test_montage.py [hunk 2]
```diff
@@ -959,14 +949,38 @@ def test_set_montage():
     raw = read_raw_fif(fif_fname)
     orig_pos = np.array([ch['loc'][:3] for ch in raw.info['chs']
                          if ch['ch_name'].startswith('EEG')])
-    raw.set_montage('mgh60')  # test loading with string argument
+    with pytest.deprecated_call():
+        raw.set_montage('mgh60')  # test loading with string argument
```
Comment (agramfort) [inline mne/channels/tests/test_montage.py]: this should still work but via the DigMontage route

Comment (massich) [reply mne/channels/tests/test_montage.py]: #6765

**Comment by massich** [comment]
Here is the rendered version of the example. Maybe we want to add couple words of why the points look a bit off some times. But this can be done later.

https://15297-1301584-gh.circle-artifacts.com/0/dev/auto_examples/visualization/plot_montage.html#sphx-glr-auto-examples-visualization-plot-montage-py

**Comment by massich** [comment]
@agramfort suggested IRL to merge this one. There #6764 #6765 should follow this one, and maybe allow to register the standard montages.

**Pull request merged**